esp-box-lite/esp-box-lite.c

```diff
@@ -24,11 +24,8 @@
 static const char *TAG = "ESP-BOX-LITE";
 
 static lv_indev_t *disp_indev = NULL;   /* Input device (buttons) */
-static const audio_codec_data_if_t *i2s_data_if = NULL;  /* Codec data interface */
-static i2s_chan_handle_t i2s_tx_chan = NULL;
-static i2s_chan_handle_t i2s_rx_chan = NULL;
 
-const button_config_t bsp_button_config[BSP_BUTTON_NUM] = {
+static const button_config_t bsp_button_config[BSP_BUTTON_NUM] = {
```

Comment:
Can the handle of `adc_oneshot_unit_handle_t *adc_handle`  be created in esp-box-lite.c and passed it to lv_port, iot_button, and this handle can be provided to users.

Otherwise, the adc button here will occupy `ADC_CHANNEL_0`, and the user cannot use other channels.

Reply:
@espressif2022 This would be nice but it is not supported in IDF4.4 and it would be be complicated to handle this API for both IDF4.4 and IDF5.

Reply:
> @espressif2022 This would be nice but it is not supported in IDF4.4 and it would be be complicated to handle this API for both IDF4.4 and IDF5.

Is there any chance that this interface will be thrown out?
I had this trouble using it, there is no way but to switch to `ADC_UNIT_2`.
ESP-IDF 4.4, It seems that this problem does not exist, because the initialization method is different.

Reply:
@espressif2022 I tried to prepare new API for ADC in BSP: https://github.com/espressif/esp-bsp/compare/9719e97c55b1cfa6c7e3bd119517e7bfde533198..5b0040adc8fdb483de4507e36d2143fb85a1cf31 could you check it please? 
@igrr @leeebo @Lzw655 Please, could you check new API for ADC initialize in BSP? This API will be available in all BSPs in future. It can help for battery voltage measurement in BSP too (e.g. Korvo-2).

Reply:
> @espressif2022 I tried to prepare new API for ADC in BSP: https://github.com/espressif/esp-bsp/compare/9719e97c55b1cfa6c7e3bd119517e7bfde533198..5b0040adc8fdb483de4507e36d2143fb85a1cf31 could you check it please? @igrr @leeebo @Lzw655 Please, could you check new API for ADC initialize in BSP? This API will be available in all BSPs in future. It can help for battery voltage measurement in BSP too (e.g. Korvo-2).

I've checked it on box-lite with esp-idf 5.0 , it's ok. User can share the `adc1_handle` now.
LGTM!